### PR TITLE
WP 5.5: default term option for taxonomies

### LINKED ISSUE #5793: Support `default_term` parameter for taxonomies
https://core.trac.wordpress.org/ticket/43517

> A new default_term argument has also been added to register_taxonomy(). Using this argument, a default term name (and optionally slug and description) can be defined. If the defined term does not exist, it will be created automatically.
>
>Additionally, a default_term_$taxonomy option will be added to the database for easy access, and parity with default_category.

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes #5793 

TODO: Field dependencies for input/text fields.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.
- [ ] My code includes PHP Unit Tests (if applicable)

### CODE REVIEW

File: classes/PodsInit.php [hunk 1]
```diff
@@ -1091,6 +1091,16 @@ public function setup_content_types( $force = false ) {
 					'sort'                  => (boolean) pods_v( 'sort', $taxonomy, false ),
 				);
 
+				// @since WP 5.5: Default terms.
+				$default_term_name = pods_v( 'default_term_name', $taxonomy, null, true );
+				if ( $default_term_name ) {
+					$pods_taxonomies[ $taxonomy_name ][ 'default_term' ] = array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:1097]: :no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: 892cd4bfa54152fa504a6ebe89f7973c</sub>

**Comment by JoryHogeveen** [comment]
@sc0ttkclark It appears that field dependencies only work for boolean types and select boxes, is that correct??

**Comment by JoryHogeveen** [comment]
@sc0ttkclark 
How is this done in Pods 2.8?
If already fixed then maybe we should address to much time to the field dependency issues on this PR

**Comment by sc0ttkclark** [comment]
Not fixed yet in 2.8 but code is moved for Pod configs in 2.8 too

**Review by sc0ttkclark — DISMISSED** [review]

**Comment by JoryHogeveen** [comment]
@sc0ttkclark This one is merged to soon, it was still in progress.

**Comment by sc0ttkclark** [comment]
@JoryHogeveen The default term functionality in this PR is tied to taxonomies, I went ahead and pulled out the options from 2.7.27 and made sure it stayed in 2.8

**Pull request merged**